spooler/spooler_test.go

```diff
@@ -0,0 +1,148 @@
+package spooler
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"testing"
+)
+
+// test with reader which has already completed
+func TestCompletedInput(t *testing.T) {
+	spooler, err := NewSpooler(strings.NewReader("this is just a test string"))
+	if err != nil {
+		t.Errorf("Cannot initialise Spooler")
+		return
```

Comment:
here and below - use `t.Fatal` instead of `t.Error + return`
```suggestion
		t.Fatalf("Cannot initialise Spooler: %v", err)
```